File: backend/app/models/favorite.py
"""Favorite models for recipe favorites feature."""

from datetime import datetime

from pydantic import BaseModel, Field


class FavoriteCreate(BaseModel):
    """Request model for creating a favorite."""

    recipe_id: str = Field(..., min_length=1, max_length=20)
    recipe_name: str = Field(..., min_length=1, max_length=255)
    recipe_thumbnail: str | None = Field(None, max_length=500)


class FavoriteResponse(BaseModel):
    """Response model for a single favorite."""

    id: str
    recipe_id: str
    recipe_name: str
    recipe_thumbnail: str | None
    created_at: datetime


class FavoriteListResponse(BaseModel):
    """Response model for list of favorites."""

    favorites: list[FavoriteResponse]
    count: int


class FavoriteStatus(BaseModel):
    """Response model for favorite status check."""

    is_favorite: bool

File: mobile/types/favorite.ts
/**
 * Favorite Types
 *
 * TypeScript types for user favorites feature.
 */

export interface Favorite {
  id: string;
  recipe_id: string;
  recipe_name: string;
  recipe_thumbnail: string | null;
  created_at: string;
}

export interface FavoriteCreate {
  recipe_id: string;
  recipe_name: string;
  recipe_thumbnail?: string | null;
}

export interface FavoriteListResponse {
  favorites: Favorite[];
  count: number;
}

export interface FavoriteStatus {
  is_favorite: boolean;
}
